airflow/models/dag.py

```diff
@@ -2173,10 +2173,10 @@ def add_tasks(self, tasks: Iterable[Operator]) -> None:
         for task in tasks:
             self.add_task(task)
 
-    def _remove_task(self, task_id: str) -> None:
+    def _remove_task(self, task: AbstractOperator) -> None:
         # This is "private" as removing could leave a hole in dependencies if done incorrectly, and this
         # doesn't guard against that
-        task = self.task_dict.pop(task_id)
+        self.task_dict.pop(task.task_id)
```

Comment:
It seems we don’t really need to change this? I like this taking the ID better since it ensures the task object is actually in the DAG. Taking the task object directly means more chance passing in a wrong value.

Reply:
We do, otherwise the tasl_group is unset on the original task (stored in the task_dict) and not the copy from prepare_for_execution

Reply:
So we’re kind of “lying” to `tg._remove()` so it removes something that’s not actually in the task group, and keeps the one we want in it? Doesn’t feel right…

I still wonder if we actually need to do `task.task_group = None` in `TaskGroup._remove()`. `DAG._remove_task()` does not set `task.dag = None`, so surely TaskGroup doesn’t really need to do that…?

Reply:
Perhaps let’s remove it anywhere and document the function that the caller should do additional cleanup as needed (and also `DAG._remove_task()` since it does not clean up `dag` in the first place). The function is “private” anyway, consistency should be less a goal if it conflicts to the general data model.